agentops/agent.py

```diff
@@ -0,0 +1,139 @@
+from .event import Event
+from .http import HttpClient
+from .helpers import safe_serialize, get_ISO_time, Providers
+from openai import Stream, AsyncStream
+from openai.types.chat import ChatCompletionChunk
+from openai.resources import AsyncCompletions
+import openai
+
+
+class Agent:
```

Comment:
We need to have a longer discussion about the implementation pattern. I'd prefer to not have strongly opinionated agent classes if we can help it-- we should try to rely on 3rd party instrumentors and lightweight decorators/context managers (`with`)